### PR TITLE
module_utils/ec2 - Add utils for the ec2_vpc_vpn* modules

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Add utils for the ec2_vpc_vpn* modules

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

module_utils/ec2

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/module_utils/ec2.py [hunk 1]
```diff
@@ -363,6 +363,48 @@ def reject_vpc_peering_connection(client, peering_id: str) -> bool:
...
+@AWSRetry.jittered_backoff()
+def delete_vpn_connection_route(client, vpn_connection_id: str, route: Dict[str, Any]) -> bool:
+    client.delete_vpn_connection_route(VpnConnectionId=vpn_connection_id, DestinationCidrBlock=route)
+    return True
+
+
+@EC2VpcErrorHandler.common_error_handler("create vpn connection")
+@AWSRetry.jittered_backoff()
+def create_vpn_connection(client, **params: Dict[str, Any]) -> Dict[str, Any]:
+    return client.create_vpn_connection(**params)
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:399]:
```suggestion
    return client.create_vpn_connection(**params)["VpnConnection"]
```

File: plugins/module_utils/ec2.py [hunk 2]
```diff
@@ -363,6 +363,48 @@ def reject_vpc_peering_connection(client, peering_id: str) -> bool:
...
+
+
+@EC2VpcErrorHandler.common_error_handler("delete vpn connection")
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:402]:
```suggestion
@EC2VpcErrorHandler.deletion_error_handler("delete vpn connection")
```

File: plugins/module_utils/ec2.py [hunk 3]
```diff
@@ -363,6 +363,48 @@ def reject_vpc_peering_connection(client, peering_id: str) -> bool:
...
+
+
+@EC2VpcErrorHandler.common_error_handler("create vpn connection route")
+@AWSRetry.jittered_backoff()
+def create_vpn_connection_route(client, vpn_connection_id: str, route: Dict[str, Any]) -> bool:
+    client.create_vpn_connection_route(VpnConnectionId=vpn_connection_id, DestinationCidrBlock=route)
+    return True
+
+
+@EC2VpcErrorHandler.common_error_handler("delete vpn connection route")
```
Comment (abikouo) [inline plugins/module_utils/ec2.py:389]:
```suggestion
@EC2VpcErrorHandler.deletion_error_handler("delete vpn connection route")
```
We should also catch the corresponding exception (`InvalidRoute.NotFound`) in the function `_is_missing`

**Review by abikouo — DISMISSED** [review]

**Review by GomathiselviS — APPROVED** [review]

**Comment by alinabuzachis** [comment]
regate

**Comment by GomathiselviS** [comment]
regate

**Review by abikouo — APPROVED** [review]

**Comment by alinabuzachis** [comment]
regate

**Comment by alinabuzachis** [comment]
regate

**Comment by alinabuzachis** [comment]
regate

**Pull request merged**